src/Tes.RunnerCLI/Commands/NodeTaskUtils.cs

```diff
@@ -31,6 +42,55 @@ public static async Task<NodeTask> DeserializeNodeTaskAsync(string tesNodeTaskFi
         }
     }
 
+    public async Task SerializeNodeTaskAsync(NodeTask nodeTask, string tesNodeTaskFilePath)
+    {
+        try
+        {
+            var nodeTaskText = JsonSerializer.Serialize(nodeTask, jsonSerializerOptions) ?? throw new InvalidOperationException("The JSON data provided is invalid.");
+
+            await File.WriteAllTextAsync(tesNodeTaskFilePath, nodeTaskText);
+        }
+        catch (Exception e)
+        {
+            Logger.LogError(e, "Failed to serialize task JSON file.");
+            throw;
+        }
+    }
+
+    public async Task<NodeTask> ResolveNodeTaskAsync(FileInfo? file, Uri? uri, Lazy<NodeTaskResolverOptions> options)
+    {
+        file?.Refresh();
+
+        if (file?.Exists ?? false)
+        {
+            return await DeserializeNodeTaskAsync(file.FullName);
+        }
+
+        ArgumentNullException.ThrowIfNull(uri);
+        ArgumentNullException.ThrowIfNull(options);
+
+        try
+        {
+            ResolutionPolicyHandler resolutionPolicy = new(options.Value.RuntimeOptions ?? throw new InvalidOperationException($"Environment variable '{nameof(NodeTaskResolverOptions)}' is missing the '{nameof(NodeTaskResolverOptions.RuntimeOptions)}' property."));
+            List<FileInput> sources = [new() { TransformationStrategy = options.Value.TransformationStrategy, SourceUrl = uri.AbsoluteUri, Path = "_" }];  // Path is not used, but it is required
+            var blobUri = (await resolutionPolicy.ApplyResolutionPolicyAsync(sources) ?? []).FirstOrDefault()?.SourceUrl ?? throw new Exception("The JSON data blob could not be resolved.");
+
+            var response = await blobApiHttpUtils.Value.ExecuteHttpRequestAsync(() => new HttpRequestMessage(HttpMethod.Get, blobUri));
+            var nodeTaskText = await response.Content.ReadAsStringAsync();
```

Comment:
This would not be retried when reading from the stream. We should use the method `ExecuteHttpRequestAndReadBodyResponseAsync`. However, for that one, you need to find out the size of the file so you can create a `PipelineBuffer` to get the data.  This is [how](https://github.com/microsoft/ga4gh-tes/blob/439fd9d5152442d5771a54809b0c223699c344a8/src/Tes.Runner/Transfer/BlobDownloader.cs#L90). This further makes the case for a wrapper around HttpBlobApiUtils, which will facilitate the unit testing of this class as well.